frontend/src/components/game/SpectatorGameView.tsx

```diff
@@ -1,18 +1,190 @@
+  }
+`;
+
+const OverflowPanel = styled(Panel)`
+  overflow-y: auto;
+  height: 100%;
+  padding: 0 25px;
+`;
+
+const NoPaddingPanel = styled(Panel)`
+  padding: 0;
+`;
 
 function SpectatorGameView() {
   const { currentUser, game } = useAppSelector((state) => state);
 
+  const [copiedEmail, setCopiedEmail] = useState(false);
   const [codeModal, setCodeModal] = useState(-1);
+  const [playerSocket, setPlayerSocket] = useState<Subscription | null>();
+  const [spectateGame, setSpectateGame] = useState<SpectateGame | null>();
+  const [error, setError] = useState<string>('');
+  const [loading, setLoading] = useState<boolean>(false);
+
+  // Unsubscribe from the player socket.
+  const unsubscribePlayer = useCallback(() => {
+    // eslint-disable-next-line no-unused-expressions
+    playerSocket?.unsubscribe();
+    setSpectateGame(null);
+  }, [playerSocket]);
+
+  // Re-subscribe in order to get the correct subscription callback.
+  const subscribePlayer = useCallback((roomIdParam: string, userIdParam: string) => {
+    setLoading(true);
+
+    // If the spectator is currently subscribed, unsubscribe them.
+    if (playerSocket) {
+      unsubscribePlayer();
+    }
+
+    // Update the spectate view based on player activity.
+    const subscribePlayerCallback = (result: Message) => {
+      if (!JSON.parse(result.body).newSpectator) {
+        const updatedSpectateGame: SpectateGame = JSON.parse(result.body);
+        setSpectateGame(updatedSpectateGame);
+      }
+    };
+
+    subscribe(routes(roomIdParam, userIdParam).subscribe_player, subscribePlayerCallback)
+      .then((subscription) => {
+        setPlayerSocket(subscription);
+
+        // Send socket message to inform player new spectator is present.
+        const newSpectator: string = JSON.stringify({
+          newSpectator: true,
+        });
+        send(routes(roomIdParam, userIdParam).subscribe_player, {}, newSpectator);
+        setError('');
+      }).catch((err) => {
+        setError(err.message);
+      }).finally(() => {
+        setLoading(false);
+      });
+  }, [playerSocket, unsubscribePlayer]);
+
+  const closeModal = () => {
+    setCodeModal(-1);
+    unsubscribePlayer();
+  };
+
+  // Creates Event when splitter bar is dragged
+  const onSecondaryPanelSizeChange = () => {
+    const event = new Event('secondaryPanelSizeChange');
+    window.dispatchEvent(event);
+  };
+
+  // Display the spectate player view if currently viewing another player.
+  if (playerSocket && spectateGame) {
+    return (
+      <>
+        <p>
+          Spectate
+          {' '}
+          {spectateGame?.player.nickname}
+        </p>
+
+        <PrimaryButton onClick={unsubscribePlayer}>
+          Go Back
+        </PrimaryButton>
+
+        {error ? <CenteredContainer><ErrorMessage message={error} /></CenteredContainer> : null}
+        <SplitterContainer>
+          <SplitterLayout
+            onSecondaryPaneSizeChange={onSecondaryPanelSizeChange}
```

Comment:
You were totally right - I tried reusing the `PlayerGameView` component, and I think it works pretty cleanly. It may be a bit more complicated to make changes in the future, but the duplicate code is reduced a good amount - thanks for the note! ✅

Reply:
That's awesome - yeah I think in the future as we keep building these two pages out it probably would be worth splitting it up again (and reducing duplicate via extracting duplicate components rather than parent-child relationship), but for now this helps reduce 100+ lines of code so it's def a plus :)